ground/gcs/src/plugins/uavobjectwidgetutils/smartsavebutton.cpp

```diff
@@ -158,7 +164,8 @@ void smartSaveButton::processOperation(QPushButton * button,bool save)
 
         if(upload_result==false) {
             qDebug() << "[smartsavebutton.cpp] Object upload error:" << obj->getName();
-            failedUploads.append(obj->getName());
+            if(mandatoryList.value(obj, true))
```

Comment:
There was a long-standing bug in Tau where attempting to turn on the autotune module would return a message about being unable to save the autotune object.  This is because there was no function to convert from display format to uavo format (e.g. milliseconds to exponential tau), but we don't want to save it anyways.

I previously fixed this at TL by setting the metadata to readonly, where it catches on the other if condition and suppressed that error.  But it's sometimes nice to be able to change that value--- either to import a uavo file with a previous set of tune measurements or to be able to override inputs to the autotune stuff.  

The mandatory list was checked for permissions but wasn't checked for these other situations.  This PR changes it back to read-write, and changes other places where savings can fail to check the mandatory list.